### PR TITLE
[support] clean up index for clause 18

### PR DESCRIPTION
**Pull request by AlisdairM**
Fix a couple of mis-indexed operators, and add index entries
for a few missing functions, most notably the new launder
function.

### CODE REVIEW

File: source/support.tex [hunk 1]
```diff
@@ -2127,7 +2127,7 @@
 terminates by throwing an exception~(\ref{expr.new}).
 \end{itemdescr}
 
-\indexlibrary{\idxcode{new}!\idxcode{operator}}%
+\indexlibrary{\idxcode{delete}!\idxcode{operator}}%
```
Comment (tkoeppe) [inline source/support.tex]: Why not array-delete?

Comment (AlisdairM) [inline source/support.tex]: I believe this is consistent with the other index entries for array-delete, and did not want to be too creative at this point.  I can consistently index 'delete[]' if preferred?

Comment (tkoeppe) [inline source/support.tex]: @zygoloid: Any preference how we index array-delete?

Comment (AlisdairM) [inline source/support.tex]: and for consistency, array-new as well.

Comment (tkoeppe) [inline source/support.tex]: Actually, don't worry. This is definitely an improvement, and it's consistent. Thanks.

**Pull request merged**